electron/main.ts

```diff
@@ -270,17 +251,147 @@ ipcMain.on('unlock', () => {
+
+ipcMain.on('openLoginPage', () => openUrlOrFile(loginUrl))
+
+ipcMain.on('openDiscordLink', () => openUrlOrFile(discordLink))
+
+ipcMain.on('openSidInfoPage', () => openUrlOrFile(sidInfoUrl))
+
+ipcMain.on('getLog', (event, appName) =>
+  openUrlOrFile(`"${heroicGamesConfigPath}/${appName}-lastPlay.log"`)
+)
+
+
+ipcMain.on('removeFolder', async (e, [path, folderName]) => {
+  if (path === 'default') {
+    const defaultInstallPath = (await GlobalConfig.get().getSettings()).defaultInstallPath.replaceAll("'", '')
+    const folderToDelete = `${defaultInstallPath}/${folderName}`
+    return setTimeout(() => {
+      exec(`rm -Rf ${folderToDelete}`)
+    }, 2000)
+  }
+
+  const folderToDelete = `${path}/${folderName}`
+  return setTimeout(() => {
+    exec(`rm -Rf ${folderToDelete}`)
+  }, 2000)
+})
+
+// Calls WineCFG or Winetricks. If is WineCFG, use the same binary as wine to launch it to dont update the prefix
+interface Tools {
+  exe: string
+  prefix: string
+  tool: string
+  wine: string
+}
+
+ipcMain.on('callTool', async (event, { tool, wine, prefix, exe }: Tools) => {
+  const wineBin = wine.replace("/proton'", "/dist/bin/wine'")
+  let winePrefix: string = prefix.replace('~', home)
+
+  if (wine.includes('proton')) {
+    const protonPrefix = winePrefix.replaceAll("'", '')
+    winePrefix = `'${protonPrefix}/pfx'`
+  }
+
+  let command = `WINE=${wineBin} WINEPREFIX=${winePrefix} 
+    ${tool === 'winecfg' ? `${wineBin} ${tool}` : tool}`
+
+  if (tool === 'runExe') {
+    command = `WINEPREFIX=${winePrefix} ${wineBin} ${exe}`
+  }
+
+  console.log({ command })
+  return exec(command)
+})
+
+/// IPC handlers begin here.
+
+ipcMain.handle('checkGameUpdates', () => LegendaryGame.checkGameUpdates())
+
+ipcMain.handle('checkVersion', () => checkForUpdates())
+
+ipcMain.handle('getMaxCpus', () => cpus().length)
+
 ipcMain.handle('getGameInfo', async (event, game) => {
   const obj = LegendaryGame.get(game)
   const info = await obj.getGameInfo()
   info.extra = await obj.getExtraInfo(info.namespace)
   return info
 })
 
+ipcMain.handle('getUserInfo', () => User.getUserInfo())
+
+// Checks if the user have logged in with Legendary already
+ipcMain.handle('isLoggedIn', () => User.isLoggedIn())
+
+ipcMain.handle('login', (event, sid) => User.login(sid))
+
+ipcMain.handle('logout', () => User.logout())
+
+ipcMain.handle('getAlternativeWine', () => GlobalConfig.get().getAlternativeWine())
+
+ipcMain.handle('readConfig', async (event, config_class) =>  {
+  switch (config_class) {
+  case 'library':
+    return await Library.get().getGames('info')
+  case 'user':
+    return User.getUserInfo().displayName
+  default:
+    console.log(`Which idiot requested '${config_class}' using readConfig?`)
+    return {}
+  }
+})
+
+ipcMain.handle('requestSettings', async (event, appName) => {
+  if (appName === 'default') {
+    return await GlobalConfig.get().config
+  }
+  return await GameConfig.get(appName).getSettings()
```

Comment:
Can we have something like `Game.get(appName).settings()` ?
Since we are requesting a settings with `get` already? 
So we could have `get.settings()` and `set.settings(settingsObj)`

Reply:
I will change that to `.config`. getSettings shouldn't be called here.

Reply:
> requesting a settings with `get` already

No we're not, we're getting the Config handler.